src/phoenix/trace/llama_index/callback.py

```diff
@@ -242,17 +247,17 @@ def on_event_end(
         **kwargs: Any,
     ) -> None:
         event_data = self._event_id_to_event_data[event_id]
-        event_data.setdefault("name", event_type.value)
-        event_data.setdefault("event_type", event_type)
-        event_data["end_event"] = CBEvent(
+        event_data.set_if_unset("name", event_type.value)
+        event_data.set_if_unset("event_type", event_type)
```

Comment:
Let's just set it (and make the existing logic simpler). Since we haven't seen any use for this extra bit of knowledge, it's not worth keeping the extra complexity around just for it.
```suggestion
        event_data.name = event_type.value
        event_data.event_type = event_type
```

Reply:
I think either way is fine - let's leave its as a :suggestion: